shell/platform/darwin/macos/framework/Source/FlutterGLCompositor.mm

```diff
@@ -0,0 +1,137 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterGLCompositor.h"
+
+#import <OpenGL/gl.h>
+
+#import "flutter/fml/logging.h"
+#import "flutter/fml/platform/darwin/cf_utils.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterBackingStoreData.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterFrameBufferProvider.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterIOSurfaceHolder.h"
+#import "third_party/skia/include/core/SkCanvas.h"
+#import "third_party/skia/include/core/SkSurface.h"
+#import "third_party/skia/include/gpu/gl/GrGLAssembleInterface.h"
+#import "third_party/skia/include/utils/mac/SkCGUtils.h"
+
+namespace flutter {
+
+FlutterGLCompositor::FlutterGLCompositor(FlutterViewController* view_controller)
+    : view_controller_(view_controller),
+      open_gl_context_(view_controller.flutterView.openGLContext) {}
+
+bool FlutterGLCompositor::CreateBackingStore(const FlutterBackingStoreConfig* config,
+                                             FlutterBackingStore* backing_store_out) {
+  CGSize size = CGSizeMake(config->size.width, config->size.height);
+
+  if (!frame_started_) {
+    StartFrame();
+    // If the backing store is for the first layer, return the fbo for the
+    // FlutterView.
+    auto fbo = [view_controller_.flutterView frameBufferIDForSize:size];
+    backing_store_out->open_gl.framebuffer.name = fbo;
+  } else {
+    FlutterFrameBufferProvider* fb_provider =
+        [[FlutterFrameBufferProvider alloc] initWithOpenGLContext:open_gl_context_];
+    FlutterIOSurfaceHolder* io_surface_holder = [FlutterIOSurfaceHolder alloc];
+
+    GLuint fbo = [fb_provider glFrameBufferId];
+    GLuint texture = [fb_provider glTextureId];
+
+    size_t layer_id = CreateCALayer();
+
+    [io_surface_holder bindSurfaceToTexture:texture fbo:fbo size:size];
+    FlutterBackingStoreData* data =
+        [[FlutterBackingStoreData alloc] initWithLayerId:layer_id
+                                              fbProvider:fb_provider
+                                         ioSurfaceHolder:io_surface_holder];
+
+    backing_store_out->open_gl.framebuffer.name = fbo;
+    backing_store_out->open_gl.framebuffer.user_data = (__bridge_retained void*)data;
+  }
+
+  backing_store_out->type = kFlutterBackingStoreTypeOpenGL;
+  backing_store_out->open_gl.type = kFlutterOpenGLTargetTypeFramebuffer;
+  backing_store_out->open_gl.framebuffer.target = GL_RGBA8;
+  backing_store_out->open_gl.framebuffer.destruction_callback = [](void* user_data) {
+    if (user_data != nullptr) {
+      CFRelease(user_data);
+    }
+  };
+
+  return true;
```

Comment:
Why did we need this if the method always return true?

Reply:
FlutterCompositor's `create_backing_store_callback` requires a bool return value.

I figured it made sense to make this method signature match so we could simply return call this function and return it's value for the callback function. Alternatively we could move the return true to the defined callback but I think this leaves more flexibility so we can return false in the case a backing store cannot be created from `FlutterGLCompositor`

```
  _compositor.create_backing_store_callback = [](const FlutterBackingStoreConfig* config,  //
                                                 FlutterBackingStore* backing_store_out,   //
                                                 void* user_data                           //
                                              ) {
    return reinterpret_cast<flutter::FlutterGLCompositor*>(user_data)->CreateBackingStore(
        config, backing_store_out);
  };
```